src/components/FormInput.jsx

```diff
@@ -0,0 +1,117 @@
+import { useRef, useState, useEffect } from "react"
+import { useNavigate } from "react-router-dom"
+import Search from "../assets/Search"
+import Discovery from "../assets/Discovery"
+import { getSearchData, getLocation } from "./fetch/api"
+import classes from "./FormInput.module.scss"
+
+const SUGGESTIONS = [
+    { id: "og", location: "Australia", capital: "Canberra" },
+    { id: "ind", location: "Indonesia", capital: "Jakarta" },
+    { id: "ph", location: "Philippines", capital: "Manila" },
+    { id: "kr", location: "Korea", capital: "Seoul" },
+    { id: "jpn", location: "Japan", capital: "tokyo" },
+]
+
+const FormInput = (props) => {
+    const { setLoading } = props
+    const locationRef = useRef()
+    const navigateTo = useNavigate()
+    const [searchData, setSearchData] = useState([])
+    const [suggestedLocations, setSuggestedLocations] = useState([])
+    const [openSuggestedLocations, setOpenSuggestedLocations] = useState(false)
+    const cityName = locationRef.current?.value
+
+    const onsearchHandler = () => {
+        if (cityName !== '') {
+            const searchObj = { cityName: locationRef.current?.value }
+            getSearchData(searchObj).then((res) => {
+                setSearchData(res)
+                setLoading(false)
+            })
+        } else {
+            return
+        }
+    }
+    const getSuggestedLocations = () => {
+        if (cityName !== '') {
+            setOpenSuggestedLocations(false)
+            getLocation(cityName).then((res) => {
+                setSuggestedLocations(res)
+            }).catch(() => {
+                setSearchData([])
+            })
+        } else {
+            return
```

Comment:
I normally put the return case first
if(!cityName) return
const searchObj ... 

No need else in this case 
Also you could look into async/await instead of .then